### DISCUSSION TITLE
Failed to send API request: error sending request for url

### DISCUSSION THREAD

**Discussion by xelamarx182** [RogueOneEcho/caesura#129] [Support]
Thanks for making this program, really excited about using it. I've been going through the initial setup and I'm receiving an error when I try to verify any torrent via URL or torrent ID (ID has been redacted in this code snippet but it is a valid torrent ID): 

`WARN △ Unsuitable for transcoding unknown`
`WARN △ Failed to send API request: error sending request for url (https://redacted.sh/ajax.php?action=torrent&id=xxxxxx)`

I've confirmed that the API key is correct. Does anyone know what I might be doing wrong?

**Comment by RogueOneEcho** [comment]
Most likely this error indicates an issue with your networking setup . Most likely DNS (in my experience). 

I presume you're using caesura via Docker? 

How familiar are you with Docker? 

Have you had any DNS or network issues with Docker containers in the past? 

Is there anything complex about your network setup? Virtual machines, Strict firewall, VPN, custom DNS etc? 

If you're comfortable doing so then I recommend `docker exec -it` into a container and running `ping` `curl`, `dig` etc to check for any network issues

**Reply by xelamarx182** [reply]
Hey thanks for the reply. I figured that it might be a networking issue. I am using caesura via Docker. I'm a novice Docker user. Have set up a few containers over the last year and am learning as I go. I have several containers running right now that are accessing the internet without issue.

I don't believe there is anything complicated about my network setup. No VPN or firewall.

I did docker exec -it into one of my running containers and ran ping, curl and dig. All seemed to work without issue so the problem does appear to be isolated to the caesura container. Unfortunately (and here is where my lack of familiarity with Docker rears its head), I don't know how to docker exec into the caesura container because the container always stops running once its process has completed and I don't know how to keep it running long enough to exec into it. Is there a setting I can enable to keep the caesura container running so I can run some tests?

Also, here's a little more info about my setup that could be contributing to the issue:

I'm running on a very old Mac Pro, and the latest version of Macos which I can have installed is v10.14. As such, I have to use an old version of Docker (19.03.13) and docker compose does not work, so I'm using run commands each time. Not sure if running an old version of Docker might create issues but figured I'd add that info.

**Reply by RogueOneEcho** [reply]
### 1. Try `--network host`

You could try running the docker commands with `--network host` that usually avoids docker networking issues.

```bash
docker run --network host [OPTIONS] IMAGE [COMMAND]
```

```bash
docker run \
--network host \
-v ./config.yml:/config.yml \
-v /path/to/your/content:/content \
-v ./output:/output \
-v ./cache:/cache \
ghcr.io/rogueoneecho/caesura \
verify 142659
```

---

### 2. Try `cargo install caesura`

Rather than using docker you could try the build steps here:

https://github.com/RogueOneEcho/caesura/blob/main/BUILD.md#install-with-cargo

Install the dependencies with `homebrew`

then `cargo install caesura`

Then run with:

```bash
caesura verify 142659
```

This is entirely untested on MacOS but it may work.

---

> Is there a setting I can enable to keep the caesura container running so I can run some tests?

No, there's no existing method that would do that.

> I'm running on a very old Mac Pro, and the latest version of Macos which I can have installed is v10.14.

Unfortunately I have no experience with MacOS and no knowledge of what limitations an earlier version of MacOS may have. My understanding is their support recommendation would be to buy a new mac...

> As such, I have to use an old version of Docker (19.03.13) and docker compose does not work, so I'm using run commands each time. Not sure if running an old version of Docker might create issues but figured I'd add that info.

Docker 19.03.13 is from 2020, so I wouldn't even consider it that old. My recollection is `docker-compose` used to be installed as a separate plugin and you would run commands via

```bash
docker-compose up -d
```

instead of 

```bash
docker compose up -d
```

So you may be able to get it running by asking ChatGPT for install instructions (or tracking down an old install guide) however I doubt this would solve the issue you're seeing, it just might make managing your docker services easier.

**Reply by xelamarx182** [reply]
Thank you so much for the detailed answer. I really appreciate it. Unfortunately --network host doesn't seem to fix the issue in the docker container so I'm giving the cargo install a try.

Cargo install seems to go fine but I am at a loss as to where my config file should be placed in this configuration. With docker I can point to the config, cache, output etc when starting the container, but I don't know how to tell the cargo-installed version where those files are. I did ask ChatGPT and tried storing the config in the various suggested defaults for config files but none of them seemed to work.

As for docker-compose, thanks for the tip. I knew the command had changed and that I would have to use docker-compose instead of docker compose but didn't realize I might need to install the separate plug-in. Again, I really appreciate your advice!

**Reply by RogueOneEcho** [reply]
> Cargo install seems to go fine but I am at a loss as to where my config file should be placed in this configuration. 

By default `caesura` expects the `config.yml` file to be in the current working directory. If you prefer to load it from somewhere else then you can use the `--config /path/to/config` option 

> Unfortunately --network host doesn't seem to fix the issue in the docker container so I'm giving the cargo install a try.

Hmm, in that case the issue may actually be with `caesura`. I've got no idea how I'd go about debugging it though.

**Reply by xelamarx182** [reply]
Thank you!! That was the last piece of the puzzle I needed. I now have caesura up and running (the version installed via Cargo) and it's working perfectly. Thanks again for your help and for making such a great program.
